napari/_qt/qt_viewer.py

```diff
@@ -443,34 +438,15 @@ def _create_canvas(self) -> None:
     def _diconnect_theme(self):
         self.viewer.events.theme.disconnect(self.canvas._on_theme_change)
 
-    def _add_visuals(self) -> None:
-        """Add visuals for axes, scale bar, and welcome text."""
+    def _add_overlay(self, overlay: Overlay):
```

Comment:
why this method is private? 
When we have a generic overlay mechanism it may be nice to expose it to the public.

Reply:
Exactly, for now we don't so I put everything private. Anyway, I think ultimately we will want to add overlays through the dict, if anything:

```py
viewer.overlays['my_overlay'] = MyOverlay(...)
```

Reply:
Remember that QtViewer is part of public API (for example for embedding). So lack of interface in `Viewer` is not blocking here.

Reply:
Still, I think it's safer to keep private for now until we figure out how that interface should look like, especially cause `_add_overlay` (just like `_add_layer`) should *stay* private, in my opinion!